File: swift/[E]283.Move-Zeroes.swift
/**
 * Runtime: 36 ms, faster than 94.81% of Swift online submissions for Move Zeroes.
 * Memory Usage: 14.6 MB, less than 95.64% of Swift online submissions for Move Zeroes.
 */
class Solution {
    func moveZeroes(_ nums: inout [Int]) {
        var index = nums.count - 1
        var zeroes = 0
        while index >= 0 {
            if nums[index] == 0 {
                nums.remove(at: index)
                zeroes += 1
            }
            index -= 1
        }
        index = 0
        while index < zeroes {
            nums.append(0)
            index += 1
        }
    }
}

File: python/[E]283.Move-Zeroes.py
# Runtime: 40 ms, faster than 98.37% of Python3 online submissions for Move Zeroes.
# Memory Usage: 15.3 MB, less than 64.36% of Python3 online submissions for Move Zeroes.
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        index = len(nums) - 1
        zeroes = 0
        while index >= 0:
            if nums[index] == 0:
                del nums[index]
                zeroes += 1
            index -= 1
        for i in range(0, zeroes):
            nums.append(0)
        
